modules/internal/DefaultRectangular.chpl

```diff
@@ -1036,13 +1036,24 @@ module DefaultRectangular {
     }
 
     proc dsiReallocate(d: domain) {
+      on this {
+      //
+      // If both d and dom are default rectangular, this is pretty
+      // easy...
+      //
       if (d._value.type == dom.type) {
-        on this {
         var copy = new DefaultRectangularArr(eltType=eltType, rank=rank,
-                                            idxType=idxType,
-                                            stridable=d._value.stridable,
-                                            dom=d._value);
-        for i in d((...dom.ranges)) do
+                                             idxType=idxType,
+                                             stridable=d._value.stridable,
+                                             dom=d._value);
+        //
+        // TODO: Making this for into a forall ought to accelerate
+        // dsiReallocate() calls, yet doing so breaks due to uint/int
+        // interaction issues today.  Deserves more of a look...
+        // Does our standalone parallel iterator not have the same
+        // type flexibility as the serial iterator?
+        //
+        for i in d[(...dom.ranges)] do
```

Comment:
I opened Jira issue 189 (https://chapel.atlassian.net/projects/CHAPEL/issues/CHAPEL-189?filter=allissues) against the forall in dsiReallocate() TODO.

To summarize what I thought I was seeing:
* with CHPL_COMM=none, I thought that for worked and that forall failed for standalone and zippered
* with CHPL_COMM=gasnet, I thought that everything was working (for, forall, zippered forall)

That seems improbable though (AFAICT, RA doesn't change its domain maps based on GASNet vs. not?), so I'm going to bed instead of verifying.  I stayed up too late last night.  Will look again with fresh eyes when I have some time.